src/Toggle/MysqlGetAllTogglesGateway.php

```diff
@@ -14,14 +14,123 @@
+        $appNotificationCopyText = $this->setDefaultForAppNotificationCopyText( $row );
+
+        return array( $screenshotUrl, $descriptionOfToggle, $descriptionOfFunctionality, $descriptionOfImplementationReason, $descriptionOfLocation, $guideUrl, $appNotificationCopyText );
+    }
+
+    /**
+     * @param $row
+     * @return string
+     */
+    protected function setDefaultForScreenshotUrl( $row )
+    {
+        $screenshotUrl = isset( $row[ 'screenshot_urls' ] ) ? $row[ 'screenshot_urls' ] : null;
+        return $screenshotUrl;
+    }
+
+    /**
+     * @param $row
+     * @return string
+     */
+    protected function setDefaultForDescriptionOfToggle( $row )
+    {
+        $descriptionOfToggle = isset( $row[ 'description_of_toggle' ] ) ? $row[ 'description_of_toggle' ] : null;
+        return $descriptionOfToggle;
+    }
+
+    /**
+     * @param $row
+     * @return string
+     */
+    protected function setDefaultForDescriptionOfFunctionality( $row )
+    {
+        $descriptionOfFunctionality = isset( $row[ 'description_of_functionality' ] ) ? $row[ 'description_of_functionality' ] : null;
+        return $descriptionOfFunctionality;
+    }
+
+    /**
+     * @param $row
+     * @return string
+     */
+    protected function setDefaultForDescriptionOfImplementationReason( $row )
+    {
+        $descriptionOfImplementationReason = isset( $row[ 'description_of_implementation_reason' ] ) ? $row[ 'description_of_implementation_reason' ] : null;
+        return $descriptionOfImplementationReason;
+    }
+
+    /**
+     * @param $row
+     * @return string
+     */
+    protected function setDefaultForDescriptionOfLocation( $row )
+    {
+        $descriptionOfLocation = isset( $row[ 'description_of_location' ] ) ? $row[ 'description_of_location' ] : null;
+        return $descriptionOfLocation;
+    }
+
+    /**
+     * @param $row
+     * @return string
+     */
+    protected function setDefaultForGuideUrl( $row )
+    {
+        $guideUrl = isset( $row[ 'guide_url' ] ) ? $row[ 'guide_url' ] : null;
+        return $guideUrl;
+    }
+
+    /**
+     * @param $row
+     * @return string
+     */
+    protected function setDefaultForAppNotificationCopyText( $row )
+    {
+        $appNotificationCopyText = isset( $row[ 'app_notification_copy_text' ] ) ? $row[ 'app_notification_copy_text' ] : null;
+        return $appNotificationCopyText;
+    }
+
+    /**
+     * @param $row
      * @return array
+     * @internal param $toggles
+     */
+    protected function getToggleFromRow( $row )
+    {
+        list( $screenshotUrl, $descriptionOfToggle, $descriptionOfFunctionality, $descriptionOfImplementationReason, $descriptionOfLocation, $guideUrl, $appNotificationCopyText ) = $this->setDefaultMarketingInformationForToggle( $row );
+        return new Toggle(
+            $row[ 'name' ],
+            $row[ 'release_id' ],
+            (bool) $row[ 'is_active' ],
+            $screenshotUrl,
```

Comment:
so instead of saving all of the various values for the marketing information into variables first why not just call the relevant get**** functions from here